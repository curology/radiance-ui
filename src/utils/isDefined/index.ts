function isDefined<T>(arg: T | null | undefined): arg is T {
  return arg !== undefined && arg !== null;
}

export { isDefined };
