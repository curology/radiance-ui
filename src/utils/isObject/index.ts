function isObject(
  val:
    | string
    | number
    | Record<string, unknown>
    | Array<unknown>
    | null
    | undefined
    | unknown,
): val is Record<string, unknown> {
  return val != null && typeof val === 'object' && !Array.isArray(val);
}

export default isObject;
