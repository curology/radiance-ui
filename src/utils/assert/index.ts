import { isDefined } from '../isDefined';

function assert(value: boolean, message?: string): asserts value;
function assert<T>(
  value: T | null | undefined,
  message?: string,
): asserts value is T;
/**
 * @see https://gist.github.com/jamiebuilds/2a40f565cba317a80bed1eec127340f6
 */
function assert(value: unknown, message?: string) {
  if (value === false || !isDefined(value)) {
    throw new Error(message ?? 'Assertion failed');
  }
}

export { assert };
