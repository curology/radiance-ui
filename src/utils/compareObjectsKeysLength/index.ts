import isObject from '../isObject';

// eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
interface RecursiveObjectType {
  [key: string]: string | number | this | null;
}

const stringifyObjects = (
  obj1: RecursiveObjectType,
  obj2: RecursiveObjectType,
) => `\n\nobj1: ${JSON.stringify(obj1)}\n\nobj2:${JSON.stringify(obj2)}`;

const compareObjectsKeysLength = (
  obj1: RecursiveObjectType,
  obj2: RecursiveObjectType,
): boolean => {
  if (Object.keys(obj1).length !== Object.keys(obj2).length) {
    console.error(`Object keys do not match:${stringifyObjects(obj1, obj2)}`);
    return false;
  }

  return Object.keys(obj1).every((key) => {
    const obj1Val = obj1[key];
    const obj2Val = obj2[key];

    // Test deep key equality recursively
    if (isObject(obj1Val) && isObject(obj2Val)) {
      return compareObjectsKeysLength(obj1Val, obj2Val);
    }

    // Check if property value is null or undefined, neither of which are valid
    // theme values, which means there was not a complementary key found
    if (obj2Val == null) {
      console.error(
        `Key "${key}" missing in obj2:${stringifyObjects(obj1, obj2)}`,
      );
      return false;
    }

    return true;
  });
};

export default compareObjectsKeysLength;
