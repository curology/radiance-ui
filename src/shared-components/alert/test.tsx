import React from 'react';
import { act, userEvent, render } from 'src/tests/testingLibraryHelpers';
import { assert } from 'src/utils/assert';

import { Alert } from './index';

// Note on truncateText prop test: this cannot be tested because element scrollHeight and offsetHeight are not simulated correctly

const alertText = 'Your email address was updated successfully!';

const CustomContentComponent = () => (
  <div>
    <strong>Your info is updated!</strong>{' '}
    <a href="curology.com"> Click here to see changes</a>
  </div>
);

describe('Alert UI snapshots', () => {
  it('renders a default alert', () => {
    const { container } = render(<Alert content={alertText} />);

    expect(container.firstElementChild).toMatchSnapshot();
  });

  it('renders success alert', () => {
    const { container } = render(<Alert content={alertText} type="success" />);

    expect(container.firstElementChild).toMatchSnapshot();
  });

  it('renders error alert', () => {
    const { container } = render(<Alert content={alertText} type="error" />);

    expect(container.firstElementChild).toMatchSnapshot();
  });

  it('renders a sticky alert', () => {
    const { container } = render(
      <Alert content={alertText} duration="sticky" />,
    );

    expect(container.firstElementChild).toMatchSnapshot();
  });

  it('renders custom component passed in content prop', () => {
    const { container } = render(
      <Alert content={<CustomContentComponent />} />,
    );

    expect(container.firstElementChild).toMatchSnapshot();
  });

  it('alert onExit is triggered on click', () => {
    jest.useFakeTimers();
    const spy = jest.fn();
    const { container } = render(
      <Alert
        content={<CustomContentComponent />}
        onExit={spy}
        duration="sticky"
      />,
    );

    assert(container.firstElementChild);
    userEvent.click(container.firstElementChild);

    act(() => {
      jest.runAllTimers();
    });

    expect(spy).toHaveBeenCalled();
  });

  it('alert with custom CTA', () => {
    jest.useFakeTimers();
    const spy = jest.fn();
    const { container } = render(
      <Alert
        content={<CustomContentComponent />}
        type="error"
        ctaContent="Update Payment Method"
        onExit={spy}
      />,
    );

    assert(container.firstElementChild);
    userEvent.click(container.firstElementChild);

    act(() => {
      jest.runAllTimers();
    });

    expect(spy).toHaveBeenCalled();
  });
});
