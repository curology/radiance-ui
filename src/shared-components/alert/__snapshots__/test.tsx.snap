// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert UI snapshots renders a default alert 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-0 {
  border: none;
  text-align: left;
  cursor: pointer;
  position: relative;
  margin: 0 auto 0.5rem;
  padding: 0;
  width: 327px;
  border-radius: 8px;
  opacity: 1;
  -webkit-animation: animation-0 350ms 1;
  animation: animation-0 350ms 1;
  -webkit-transition: 350ms;
  transition: 350ms;
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  transform: translate3d(0, 0, 0);
  background-color: #332E54;
  box-shadow: 0px 8px 24px rgba(51, 46, 84, 0.05);
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

@media (min-width: 768px) {
  .emotion-0 {
    -webkit-animation: animation-1 350ms 1;
    animation: animation-1 350ms 1;
    margin: 0 auto 1rem;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #FFFFFF;
  font-size: 0.875rem;
  padding: 1rem;
}

.emotion-4 svg {
  height: 1rem;
  width: 1rem;
  fill: #FFFFFF;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #FFFFFF;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  margin: -3px 0 0 1rem;
  max-height: none;
}

<button
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="#FFFFFF"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.52 2a1 1 0 100-2 1 1 0 000 2zM8.55 14.73V4.74L6 4.73v1.25h1.3v8.75H6v1.25h3.85v-1.25h-1.3z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
    >
      Your email address was updated successfully!
    </div>
  </div>
</button>
`;

exports[`Alert UI snapshots renders a sticky alert 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-0 {
  border: none;
  text-align: left;
  cursor: pointer;
  position: relative;
  margin: 0 auto 0.5rem;
  padding: 0;
  width: 327px;
  border-radius: 8px;
  opacity: 1;
  -webkit-animation: animation-0 350ms 1;
  animation: animation-0 350ms 1;
  -webkit-transition: 350ms;
  transition: 350ms;
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  transform: translate3d(0, 0, 0);
  background-color: #332E54;
  box-shadow: 0px 8px 24px rgba(51, 46, 84, 0.05);
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

@media (min-width: 768px) {
  .emotion-0 {
    -webkit-animation: animation-1 350ms 1;
    animation: animation-1 350ms 1;
    margin: 0 auto 1rem;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #FFFFFF;
  font-size: 0.875rem;
  padding: 1rem;
}

.emotion-4 svg {
  height: 1rem;
  width: 1rem;
  fill: #FFFFFF;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #FFFFFF;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  margin: -3px 0 0 1rem;
  max-height: none;
}

<button
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="#FFFFFF"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.52 2a1 1 0 100-2 1 1 0 000 2zM8.55 14.73V4.74L6 4.73v1.25h1.3v8.75H6v1.25h3.85v-1.25h-1.3z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
    >
      Your email address was updated successfully!
    </div>
  </div>
</button>
`;

exports[`Alert UI snapshots renders custom component passed in content prop 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-0 {
  border: none;
  text-align: left;
  cursor: pointer;
  position: relative;
  margin: 0 auto 0.5rem;
  padding: 0;
  width: 327px;
  border-radius: 8px;
  opacity: 1;
  -webkit-animation: animation-0 350ms 1;
  animation: animation-0 350ms 1;
  -webkit-transition: 350ms;
  transition: 350ms;
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  transform: translate3d(0, 0, 0);
  background-color: #332E54;
  box-shadow: 0px 8px 24px rgba(51, 46, 84, 0.05);
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

@media (min-width: 768px) {
  .emotion-0 {
    -webkit-animation: animation-1 350ms 1;
    animation: animation-1 350ms 1;
    margin: 0 auto 1rem;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #FFFFFF;
  font-size: 0.875rem;
  padding: 1rem;
}

.emotion-4 svg {
  height: 1rem;
  width: 1rem;
  fill: #FFFFFF;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #FFFFFF;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  margin: -3px 0 0 1rem;
  max-height: none;
}

<button
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="#FFFFFF"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.52 2a1 1 0 100-2 1 1 0 000 2zM8.55 14.73V4.74L6 4.73v1.25h1.3v8.75H6v1.25h3.85v-1.25h-1.3z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
    >
      <div>
        <strong>
          Your info is updated!
        </strong>
         
        <a
          href="curology.com"
        >
           Click here to see changes
        </a>
      </div>
    </div>
  </div>
</button>
`;

exports[`Alert UI snapshots renders error alert 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-0 {
  border: none;
  text-align: left;
  cursor: pointer;
  position: relative;
  margin: 0 auto 0.5rem;
  padding: 0;
  width: 327px;
  border-radius: 8px;
  opacity: 1;
  -webkit-animation: animation-0 350ms 1;
  animation: animation-0 350ms 1;
  -webkit-transition: 350ms;
  transition: 350ms;
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  transform: translate3d(0, 0, 0);
  background-color: #BD200F;
  box-shadow: 0px 8px 24px rgba(189, 32, 15, 0.05);
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

@media (min-width: 768px) {
  .emotion-0 {
    -webkit-animation: animation-1 350ms 1;
    animation: animation-1 350ms 1;
    margin: 0 auto 1rem;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #FFFFFF;
  font-size: 0.875rem;
  padding: 1rem;
}

.emotion-4 svg {
  height: 1rem;
  width: 1rem;
  fill: #FFFFFF;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #FFFFFF;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  margin: -3px 0 0 1rem;
  max-height: none;
}

<button
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="#FFFFFF"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M14.6 6.79a6.747 6.747 0 00-7.9-5.36 6.7 6.7 0 00-4.31 2.84 6.71 6.71 0 00-1.04 5.06 6.747 6.747 0 007.9 5.36 6.7 6.7 0 004.31-2.84 6.71 6.71 0 001.04-5.06zM6.46.2C6.97.11 7.48.06 7.98.06a7.996 7.996 0 011.5 15.85c-.51.09-1.02.14-1.52.14-3.76 0-7.12-2.66-7.85-6.49A7.996 7.996 0 016.46.2zm.86 3.91h1.41v5.26H7.32V4.11zm-.17 6.83c0-.48.39-.84.88-.84s.88.36.88.84c0 .49-.39.86-.88.86s-.88-.37-.88-.86z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
    >
      Your email address was updated successfully!
    </div>
  </div>
</button>
`;

exports[`Alert UI snapshots renders success alert 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-0 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -24px, 0);
    -moz-transform: translate3d(0, -24px, 0);
    -ms-transform: translate3d(0, -24px, 0);
    transform: translate3d(0, -24px, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0px, 0);
    -moz-transform: translate3d(0, 0px, 0);
    -ms-transform: translate3d(0, 0px, 0);
    transform: translate3d(0, 0px, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes animation-1 {
  from {
    opacity: 0;
    -webkit-transform: translate3d(24px, 0, 0);
    -moz-transform: translate3d(24px, 0, 0);
    -ms-transform: translate3d(24px, 0, 0);
    transform: translate3d(24px, 0, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-0 {
  border: none;
  text-align: left;
  cursor: pointer;
  position: relative;
  margin: 0 auto 0.5rem;
  padding: 0;
  width: 327px;
  border-radius: 8px;
  opacity: 1;
  -webkit-animation: animation-0 350ms 1;
  animation: animation-0 350ms 1;
  -webkit-transition: 350ms;
  transition: 350ms;
  -webkit-transform: translate3d(0, 0, 0);
  -moz-transform: translate3d(0, 0, 0);
  -ms-transform: translate3d(0, 0, 0);
  transform: translate3d(0, 0, 0);
  background-color: #2B6E33;
  box-shadow: 0px 8px 24px rgba(43, 110, 51, 0.05);
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

@media (min-width: 768px) {
  .emotion-0 {
    -webkit-animation: animation-1 350ms 1;
    animation: animation-1 350ms 1;
    margin: 0 auto 1rem;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #FFFFFF;
  font-size: 0.875rem;
  padding: 1rem;
}

.emotion-4 svg {
  height: 1rem;
  width: 1rem;
  fill: #FFFFFF;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #FFFFFF;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  margin: -3px 0 0 1rem;
  max-height: none;
}

<button
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="#FFFFFF"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.96 2l-9.7 9.42L.88 7.1 0 7.99l5.26 5.18L15.83 2.9l-.87-.9z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
    >
      Your email address was updated successfully!
    </div>
  </div>
</button>
`;
